### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by emartynov**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description

Fix warious compile warning

## :bulb: Motivation and Context

I like to have the compilation log without warnings.

## :green_heart: How did you test it?
Compile and run the application.

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code

## :crystal_ball: Next steps
Not all warnings are fixed but rest will require some rethinking.

Partial fix for #745

### CODE REVIEW

**Comment by emartynov** [comment]
I experienced a couple of unsuccessful builds but message was not clear - gradle was not able to connect to child process:

```
Unable to connect to the child process 'Gradle Test Executor 8'.
  It is likely that the child process have crashed - please find the stack trace in the build log.
  This exception might occur when the build machine is extremely loaded.
  The connection attempt hit a timeout after 120.0 seconds (last known process state: STARTED, running: true).
```

I wonder if it is connected to my changes or just circle ci is busy.

**Comment by florina-muntenescu** [comment]
could you fix the conflicts, please?

**Comment by emartynov** [comment]
@florina-muntenescu done!

File: core/src/main/java/io/plaidapp/core/data/DataManager.kt [hunk 1]
```diff
@@ -82,7 +83,9 @@ class DataManager @Inject constructor(
     init {
         sourcesRepository.registerFilterChangedCallback(filterListener)
         // build a map of source keys to pages initialized to 0
-        pageIndexes = sourcesRepository.getSourcesSync().map { it.key to 0 }.toMap().toMutableMap()
+        runBlocking {
```
Comment (florina-muntenescu) [inline core/src/main/java/io/plaidapp/core/data/DataManager.kt]: I'd rather not use `runBlocking` in non-tests code. Given that we want to re-write this at one point, let's leave this as it was, please

Comment (emartynov) [reply core/src/main/java/io/plaidapp/core/data/DataManager.kt]: Rebased my fork on latest changes and returned warning back as you asked

File: core/src/main/java/io/plaidapp/core/designernews/domain/LoadStoriesUseCase.kt [hunk 2]
```diff
@@ -32,14 +31,14 @@ class LoadStoriesUseCase @Inject constructor(
 
     suspend operator fun invoke(page: Int): Result<List<Story>> {
```
Comment (ashdavies) [inline core/src/main/java/io/plaidapp/core/designernews/domain/LoadStoriesUseCase.kt]: I noticed this and was puzzled as I didn't think `Result` could be used as a return type?

Comment (florina-muntenescu) [reply core/src/main/java/io/plaidapp/core/designernews/domain/LoadStoriesUseCase.kt]: We are doing this in several places in Plaid.

**Pull request merged**